forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -40,30 +87,66 @@ pub async fn deploy(command: cmd::Deploy) -> Result<Vec<DeployedContract>> {
     let build_opts = build_opts_from_cmd(&command);
     let built_pkgs_with_manifest = built_pkgs_with_manifest(&curr_dir, build_opts)?;
 
-    // If a salt was specified but we have more than one member to build, there
-    // may be ambiguity in how the salt should be applied, especially if the
-    // workspace contains multiple contracts, and especially if one contract
-    // member is the dependency of another (in which case salt should be
-    // specified under `[contract- dependencies]`). Considering this, we have a
-    // simple check to ensure that we only accept salt when deploying a single
-    // package. In the future, we can consider relaxing this to allow for
-    // specifying a salt for workspace deployment, as long as there is only one
-    // root contract member in the package graph.
-    if command.salt.salt.is_some() && built_pkgs_with_manifest.len() > 1 {
-        bail!(
-            "A salt was specified when attempting to deploy a workspace with more than one member.
-              If you wish to deploy a contract member with salt, deploy the member individually.
-              If you wish to specify the salt for a contract dependency, \
-            please do so within the `[contract-dependencies]` table."
-        )
-    }
+    let contract_salt_map = if let Some(salt_input) = &command.salt {
+        // If we're building 1 package, we just parse the salt as a string, ie. 0x00...
+        // If we're building >1 package, we must parse the salt as a pair of strings, ie. contract_name:0x00...
+        if built_pkgs_with_manifest.len() > 1 {
+            let map = validate_and_parse_salts(
+                salt_input.clone(),
+                built_pkgs_with_manifest
+                    .iter()
+                    .map(|bpwm| bpwm.package_manifest_file()),
+            )?;
+
+            Some(map)
+        } else {
+            if salt_input.len() > 1 {
+                bail!("More than 1 salt was specified when deploying a single contract");
+            }
+
+            // OK to index into salt_input ahd built_pkgs_with_manifest here,
```

Comment:
Do we need to handle the case where `built_pkgs_with_manifest.len() == 0`?

Reply:
This shouldn't be able to return 0 technically, I guess adding a check upstream within built_pkgs_with_manifest would be good here